lib/timers.js

```diff
@@ -496,7 +501,15 @@ function rearm(timer) {
     timer._handle.start(timer._repeat);
   } else {
     timer._idleTimeout = timer._repeat;
-    active(timer);
+
+    const duration = TimerWrap.now() - start;
+    if (duration >= timer._repeat) {
+      // If callback duration >= timer._repeat,
+      // add 1 ms to avoid blocking eventloop
+      process.nextTick(insert, timer, false, start + duration - timer._repeat);
+    } else {
+      insert(timer, false, start);
+    }
```

Comment:
Yes, the conditional check is needed if callback > interval, otherwise, loop will be blocked.